### DISCUSSION TITLE
Separate+combine in Oklab/Oklch colorspaces?

### DISCUSSION THREAD

**Discussion by vrzlev** [ImageMagick/ImageMagick#6443] [Help]
### ImageMagick version

7.1.1

### Operating system, version and so on

openSUSE Tumbleweed

### Description

Hello!

Trying to learn about colorspaces, hit this obstacle with `Oklch`:

```
magick -size 800x800 plasma: plasma1.png
magick plasma1.png -colorspace Oklch -separate lch%d.png
magick lch0.png lch1.png lch2.png -set colorspace Oklch -combine -colorspace sRGB plasma2.png
```

Expected to roughly restore the original input, but the result is a completely black rectangle. Same happens in `Oklab` colorspace.

Any guidance is appreciated. Thank you!

**Comment by snibgo** [comment]
You write to png files, but read from jpg files.

**Reply by vrzlev** [reply]
Sorry, wrong copy-paste from bash history. Updated the snippet. I did in fact try many variations, but none worked.

**Comment by snibgo** [comment]
Those colorspaces are very new, and not in the versions I have, so I can't comment, sorry. Perhaps they have bugs.

**Comment by fmw42** [comment]
Your command should have -colorspace sRGB after -combine.  Nevertheless, it does not work for me either on Mac OSX Ventura and IM 7.1.1.12

Looks like a bug to me, also.

**Reply by vrzlev** [reply]
Oops, sorry about that. Same reporting mistake, it was 2am. ;(

**Reply by vrzlev** [reply]
Filed as https://github.com/ImageMagick/ImageMagick/issues/6444.
